src/Compilers/Core/Portable/Emit/CommonPEModuleBuilder.cs

```diff
@@ -421,6 +421,36 @@ internal abstract class PEModuleBuilder<TCompilation, TSourceModuleSymbol, TAsse
 
         public abstract TEmbeddedTypesManager EmbeddedTypesManagerOpt { get; }
 
+        private bool _needsNonNullTypesAttribute;
+        /// <summary>
+        /// Track usage of the NonNullTypesAttribute injected type, so it can be added to module to be emitted.
+        /// </summary>
+        protected bool NeedsNonNullTypesAttribute
+        {
+            get => _needsNonNullTypesAttribute;
```

Comment:
>_needsNonNullTypesAttribute [](start = 19, length = 27)

Is it Ok to query this property before InjectedSymbolsAreFrozen?

Reply:
Yes. The method that does the freezing will query those two values (and may set the second one too).

---
In reply to: [215084482](https://github.com/dotnet/roslyn/pull/29180#discussion_r215084482) [](ancestors = 215084482)

Reply:
> Yes. The method that does the freezing will query those two values (and may set the second one too).

Could other consumers be mislead by getting not-final value?

---
In reply to: [215093886](https://github.com/dotnet/roslyn/pull/29180#discussion_r215093886) [](ancestors = 215093886,215084482)

Reply:
I think the patters that we were trying to follow is either asking for a value calculates the final value, or one shouldn't be asking for the value until we are ready to give the final result.

---
In reply to: [215094274](https://github.com/dotnet/roslyn/pull/29180#discussion_r215094274) [](ancestors = 215094274,215093886,215084482)

Reply:
There is only one consumer.
By moving the storage to C# assembly builder (per comment above), I can remove the property, which I think will resolve the concern anyways.

---
In reply to: [215094274](https://github.com/dotnet/roslyn/pull/29180#discussion_r215094274) [](ancestors = 215094274,215093886,215084482)